### PR TITLE
cli: remove `cockroach quit`

### PR DESCRIPTION
**Pull request by knz**
As per https://github.com/cockroachdb/docs/pull/14299.

Release note (backward-incompatible change): The `cockroach quit`
command was removed. It had been deprecated since v20.1. To shut down
a node gracefully, send a SIGTERM signal to it.

cc @taroface

### CODE REVIEW

**Comment by erikgrinaker** [comment]
>Unsure if this needs to be documented - is roachprod public-facing?

No, `roachprod` isn't public-facing, so we can skip the release note.

**Comment by srosenberg** [comment]
> > Unsure if this needs to be documented - is roachprod public-facing?
> 
> No, `roachprod` isn't public-facing, so we can skip the release note.

Technically it is; e.g., cloud report playbook uses it.

File: build/archive/contents/README [hunk 1]
```diff
@@ -19,7 +19,7 @@ To interact with a one-node SQL cluster:
 
      $ cockroach start --background --insecure
      $ cockroach sql --insecure
-     $ cockroach quit --insecure
+     $ killall -TERM cockroach # to shut down the nodes
```
Comment (erikgrinaker) [inline build/archive/contents/README:22]: One thing to note here is that on some Unices (looking at you Solaris), `killall` will kill _all_ processes on the system. Maybe that's obscure enough that we don't have to worry about it, given that we only officially support Linux and macOS does something similar here. But worth keeping in mind if we're going to sprinkle this all over our docs etc.

**Review by erikgrinaker — APPROVED** [review]
LGTM at a high level.

**Comment by erikgrinaker** [comment]
> > > Unsure if this needs to be documented - is roachprod public-facing?
> > 
> > 
> > No, `roachprod` isn't public-facing, so we can skip the release note.
> 
> Technically it is; e.g., cloud report playbook uses it.

Doesn't it make certain assumptions about our cloud setup though, e.g. project names and VM names and stuff? Seems like more trouble than it's worth to officially support it.

File: build/archive/contents/README (hunk 1 above)
Comment (srosenberg) [reply build/archive/contents/README:22]: Good point, Erik! Personally, I never run `killall`. In this case, I'd probably use `pgrep cockroach | xargs kill -TERM`, but I am also fine with `killall` being in README.

File: build/verify-archive.sh [hunk 2]
```diff
@@ -9,7 +9,7 @@
 set -euo pipefail
 
 apt-get update
-apt-get install -y autoconf bison cmake libncurses-dev
+apt-get install -y autoconf bison cmake libncurses-dev psmisc
```
Comment (srosenberg) [inline build/verify-archive.sh:12]: I assume that's for `killall`? Maybe we could do away by using `pgrep cockroach | kill -TERM` instead?

File: pkg/cli/flags_test.go [hunk 3]
```diff
@@ -962,32 +962,33 @@ func TestClientConnSettings(t *testing.T) {
 
 	// For some reason, when run under stress all these test cases fail due to the
 	// `--host` flag being unknown to quitCmd. Just skip this under stress.
+	// TODO(knz): Check if this skip still applies.
```
Comment (srosenberg) [inline pkg/cli/flags_test.go:965]: Would be nice to lift the `skip` if this no longer holds.

**Review by knz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/82988)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @erikgrinaker, @srosenberg, and @tbg)

___
*[`build/verify-archive.sh` line 12 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/82988#-N4gcb0Vm8_DMmV5BnMy-r7-12:-N4hF4B94CUbnFqrmfQ1:b-lm1kqy) ([raw file](https://github.com/cockroachdb/cockroach/blob/fe0792aa9ef98930d5c45be02b2524a646c241ac/build/verify-archive.sh#L12)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

I assume that's for `killall`? Maybe we could do away by using `pgrep cockroach | kill -TERM` instead?
</blockquote></details>

`pgrep` is part of `procps` which needs to be installed to. Changed to use that instead.
___
*[`build/archive/contents/README` line 22 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/82988#-N4gcb0Vm8_DMmV5BnMx-r6-22:-N4hFB2y-pCRAvi1JKs0:b8rdukw) ([raw file](https://github.com/cockroachdb/cockroach/blob/cf107029fb364ef8aca94fa2a8d6450256c40ec0/build/archive/contents/README#L22)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

Good point, Erik! Personally, I never run `killall`. In this case, I'd probably use `pgrep cockroach | xargs kill -TERM`, but I am also fine with `killall` being in README.
</blockquote></details>

Changed to `pkill`.
___
*[`pkg/cli/flags_test.go` line 965 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/82988#-N4gcb0Vm8_DMmV5BnMz-r7-965:-N4hFCgj-eK863JgRdN9:b-pdt05y) ([raw file](https://github.com/cockroachdb/cockroach/blob/fe0792aa9ef98930d5c45be02b2524a646c241ac/pkg/cli/flags_test.go#L965)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

Would be nice to lift the `skip` if this no longer holds.
</blockquote></details>

I agree. But I don't think this should hold this PR.

**Comment by knz** [comment]
bors r=erikgrinaker,srosenberg

**Comment by knz** [comment]
@tbg points out this might be flaky. Will investigate further.

bors r-

**Comment by knz** [comment]
bors r=erikgrinaker,srosenberg

**Pull request merged**